Sming/SmingCore/SdkCompat.h

```diff
@@ -0,0 +1,32 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/anakod/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ ****/
+
+#ifndef SDKCOMPAT_H_
+#define SDKCOMPAT_H_
+
+#ifndef PERIPHS_IO_MUX_PULLDWN
+#define PERIPHS_IO_MUX_PULLDWN          BIT6
+#endif
+
+#ifndef PIN_PULLDWN_DIS
+#define PIN_PULLDWN_DIS(PIN_NAME)             CLEAR_PERI_REG_MASK(PIN_NAME, PERIPHS_IO_MUX_PULLDWN)
+#endif
+
+#ifndef PIN_PULLDWN_EN
+#define PIN_PULLDWN_EN(PIN_NAME)              SET_PERI_REG_MASK(PIN_NAME, PERIPHS_IO_MUX_PULLDWN)
+#endif
+
+#ifndef os_intr_lock
+#define os_intr_lock ets_intr_lock
+#endif
+
+#ifndef os_intr_unlock
+#define os_intr_unlock ets_intr_unlock
+#endif
+
+#endif /* SDKCOMPAT_H_ */
```

Comment:
> I think this file should be removed, because pull-down isn't accessible in hardware (is it right, yes?). 

According to Espressif the hardware does not support this. I would say that we should listen to them and do as you suggested - remove all occurences of PULL_DOWN. 

As for the os_intr_unclock and os_intra_lock if you want I can replace them directly in the framework. 

And finally I can remove them SdkCompat.h because it won't be needed any longer.

What do you think? Comments from your side?